src/signalcorr.jl

```diff
@@ -61,34 +59,31 @@ where each column in the result will correspond to a column in `x`.
 
 The output is not normalized. See [`autocor!`](@ref) for a method with normalization.
 """
-function autocov!(r::RealVector, x::RealVector, lags::IntegerVector; demean::Bool=true)
+function autocov!(
+    r::AbstractVector,
+    x::AbstractVector,
+    lags::AbstractVector{<:Integer},
+    z::Vector=Vector{typeof(zero(eltype(x)) / 1)}(undef, length(x));
```

Comment:
@kagalenko-m-b What do you think about adding an internal `_autocov` function that requires a `z` vector and could be used by the different `autocov!` and `autocor!` methods? Then the `z` cache would not leak to the user-facing API and still code duplications could be reduced.